mautic/mautic
# A/B tests various fixes

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:
This PR cover some fixes for A/B testing
We've noticed some issues:

- switch between variant and all does not work properly 
![image](https://user-images.githubusercontent.com/462477/216513372-fb1eccd5-ec92-40b2-aa2b-5fbf992081b7.png)

- percentage and graphs doe don't work If one of email is unscheduled or unpublished 

![image](https://user-images.githubusercontent.com/462477/216513531-a39c187e-7625-43de-972f-7a4266089245.png)

- send by broadcast send also children emails, even IF we already send some emails during parent email sending
 



#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2.  Create A/B testing scheduled  emails with 100 contacts for example
3.  Send it by broadcast command  `mautic:broadcast:send --limit=10` 
4.  See like your command send 20 emails (we expect 10 split 50% for each)
5. Check graphs and see like email was not send divided 50% for each email, there is diffference
6. Wait until one of the email is unscheduled or unpublished
7. Then check graphs na percentage in the A/B tabs on detail email page
8. We expect see graph and correct percentage
9. After apply PR should all these issues fixed
